staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -72,6 +73,29 @@ var (
 	errExpectSliceItems = errors.New("Items field must be a slice of objects")
 )
 
+func GetListMeta(list runtime.Object) (metav1.TypeMeta, metav1.ListMeta, []runtime.Object, error) {
+	v, err := conversion.EnforcePtr(list)
+	if err != nil {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, err
+	}
+	listField := v.FieldByName("ListMeta")
+	if !listField.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta")
+	}
+	listMeta := listField.Interface().(metav1.ListMeta)
+
```

Comment:
should this also ensure the list meta field has a json tag encoding as `metadata`?

Reply:
Sure, as that would make GetListMeta function json specific I would move it.